deepcell/image_generators/semantic.py

```diff
@@ -385,3 +384,527 @@ def random_transform(self, x, y=None, seed=None):
 
         self.interpolation_order = _interpolation_order
         return x, y
+
+
+class SemanticMovieIterator(Iterator):
+    """Iterator yielding data from Numpy arrays (X and y).
+
+    Args:
+        train_dict (dict): Dictionary consisting of numpy arrays for X and y.
+        movie_data_generator: Instance of SemanticMovieGenerator
+            to use for random transformations and normalization.
+        batch_size (int): Size of a batch.
+        shuffle (boolean): Whether to shuffle the data between epochs.
+        seed (int): Random seed for data shuffling.
+        data_format (str): One of 'channels_first', 'channels_last'.
+        save_to_dir (str): Optional directory where to save the pictures
+            being yielded, in a viewable format. This is useful
+            for visualizing the random transformations being
+            applied, for debugging purposes.
+        save_prefix (str): Prefix to use for saving sample
+            images (if save_to_dir is set).
+        save_format (str): Format to use for saving sample images
+            (if save_to_dir is set).
+
+    Returns:
+        Iterator yielding data X and y, where y is a list containing
+        transformed masks. Data is randomly augmented using
+        image_data_generator.
+    """
+
+    def __init__(self,
+                 train_dict,
+                 movie_data_generator,
+                 batch_size=1,
+                 frames_per_batch=5,
+                 shuffle=False,
+                 transforms=['watershed-cont'],
+                 transforms_kwargs={},
+                 seed=None,
+                 min_objects=3,
+                 data_format='channels_last',
+                 save_to_dir=None,
+                 save_prefix='',
+                 save_format='png'):
+        X, y = train_dict['X'], train_dict['y']
+        if X.shape[0] != y.shape[0]:
+            raise ValueError('Training batches and labels should have the same'
+                             'length. Found X.shape: {} y.shape: {}'.format(
+                                 X.shape, y.shape))
+
+        if X.ndim != 5:
+            raise ValueError('Input data in `SemanticMovieIterator` '
+                             'should have rank 5. You passed an array '
+                             'with shape', X.shape)
+
+        if y is None:
+            raise ValueError('Instance masks are required for the '
+                             'SemanticIterator')
+
```

Comment:
I think the code will have a `KeyError` when we call  `X, y = train_dict['X'], train_dict['y']`, and this `y is None line will never get called.